app/models/user.rb

```diff
@@ -21,18 +21,24 @@ def queued_video?(video)
     queue_items.map(&:video).include?(video)
   end
 
-  def follows?(another_user)
-    following_relationships.map(&:leader).include? another_user
+  def follow(leader)
+    if self.can_follow?(leader)
+      following_relationships.create(leader_id: leader.id)
+    end
   end
 
-  def can_follow?(another_user)
-    !(self.follows?(another_user) || another_user == self)
+  def follows?(another_user)
+    following_relationships.map(&:leader).include? another_user
```

Comment:
Another, more performant, way to do this is:

```ruby
following_relationships.pluck(:leader_id).include? another_user.id
```

This doesn't require loading all of the models in memory at once.